msyncd/synchronizer.cpp

```diff
@@ -1903,10 +1900,22 @@ QStringList Synchronizer::syncProfilesByType(const QString &aType)
     return iProfileManager.profileNames(aType);
 }
 
-void Synchronizer::onNetworkStateChanged(bool aState)
+void Synchronizer::onNetworkStateChanged(bool aState, Sync::InternetConnectionType type)
 {
     FUNCTION_CALL_TRACE;
-    if(!aState) {
+    LOG_DEBUG("Network state changed: OnLine:" << aState << " connection type:" <<  type);
+    if (acceptScheduledSync(aState, type))
+    {
+        LOG_DEBUG("Restart sync for profiles that need network");
+        QStringList profiles(iWaitingOnlineSyncs);
+        iWaitingOnlineSyncs.clear();
+        foreach(QString profileName, profiles)
+        {
+            startSync(profileName, true);
```

Comment:
Yes all 'waiting online' are scheduled syncs, manually sync triggered while offline will return a network error. And as mentioned above the 'startScheduledSync' only checks for network connection before call 'startSync' and since we already knows that a connection is available for sync there is no need to go through the check again.